rslib/src/import_export/package/apkg/export.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{
+    collections::HashSet,
+    path::{Path, PathBuf},
+};
+
+use tempfile::NamedTempFile;
+
+use crate::{
+    collection::CollectionBuilder,
+    import_export::{
+        gather::ExchangeData,
+        package::{
+            colpkg::export::{export_collection, MediaIter},
+            Meta,
+        },
+    },
+    io::{atomic_rename, tempfile_in_parent_of},
+    prelude::*,
+    storage::SchemaVersion,
+};
+
+impl Collection {
+    /// Returns number of exported notes.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Should we add a new options struct to avoid this lint, or pass the proto message directly?

Reply:
I can live with that warning for now, but if you'd like to avoid it, the native struct might be better, as then we wouldn't need to divide the args up between the proto struct and separate ones, and could make a builder. At one point it might be worth looking into something like https://crates.io/crates/derive_builder as well - but probably better not in this PR :-)